r/R/dataset-scan.R

```diff
@@ -185,17 +183,36 @@ ScanTask <- R6Class("ScanTask",
 #' `data.frame`? Default `TRUE`
 #' @export
 map_batches <- function(X, FUN, ..., .data.frame = TRUE) {
-  if (.data.frame) {
-    lapply <- map_dfr
-  }
-  scanner <- Scanner$create(ensure_group_vars(X))
+  # TODO: possibly refactor do_exec_plan to return a RecordBatchReader
```

Comment:
Would you mind making the Jira for this + put the number here? I don't know of one off the top of my head but we should get one if we think we'll (possibly) want to move in that direction

Reply:
JIra created: https://issues.apache.org/jira/browse/ARROW-15271

Added to the comment as well.